### ISSUE TITLE
In which config (or in None) should the `predict_epsilon` config be saved?

### ISSUE THREAD

**Issue by natolambert** [huggingface/diffusers#29]
We're discussing where certain properties should be saved in this repository. For example, [this `predict_epsilon`](https://github.com/huggingface/diffusers/blob/320506c75ad7d40c7d9fc6db5a45955906c5f729/src/diffusers/schedulers/scheduling_ddpm.py#L125) property is used to distinguish if a model is trained to directly reconstruct the original sample or the noise associated with it.

(used in this [colab](https://colab.research.google.com/drive/1TmBmlYeKUZSkUZoJqfBmaicVTKx6nN1R?usp=sharing))

There are multiple arguments for where it could fall, and it is an interesting discussion when creating a library that is modular and useful at different levels of abstraction. Here are the different places the `predict_epsilon` property could be stored
* **model config**: here the user could load a pre-trained model and instantly know to use it. Though, there is an argument to be made that the best model classes will be data-agnostic and this is related to the training structure.
* **pipeline config**: here the config will automatically handle the sampling, likely without the user knowing, and it will happen naturally. This is likely best if models are going to be used primarily for inference 
* **scheduler config**: this is the last natural option and operates at a sort of conceptual middle ground. The scheduler is where is sampling actually occurs, so maybe that is the best spot for the config.

Important context is that the library is currently heavily used for inference, but hopefully will be used for training as well. Some of the positioning of things like `predict_epsilon` will be more logical in a training script rather than inference.

**Comment by patrickvonplaten** [comment]
@patil-suraj @anton-l what do you think?

**Comment by patrickvonplaten** [comment]
Let's close this for now and discuss on PR once we go multi-modality
